src/lib/config.rs

```diff
@@ -0,0 +1,119 @@
+use std::{
+    env,
+    fs::{canonicalize, read_to_string},
+    io::{stderr, Write},
+    path::Path,
+    process,
+};
+
+use getopts::Options;
+
+pub struct Config {
+    /// The maximum number of parallel jobs to run.
+    pub maxjobs: usize,
+    /// The port to listen on.
+    pub port: u16,
+    /// A *fully canonicalised* path to the directory containing per-repo programs.
+    pub reposdir: String,
+    /// The GitHub secret used to validate requests.
+    pub secret: String,
```

Comment:
I'm not suggesting you do this if you don't want, but I like to use `secstr` for in-memory secrets:
https://crates.io/crates/secstr

Reply:
OK, might make sense for a future PR.